pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -104,37 +109,64 @@ func (e shortcutExpander) getShortcutMappings() ([]kubectl.ResourceShortcuts, er
 	return res, nil
 }
 
+// resourceFinder will find the resources through discovery client as per resource plura name and singular name.
+func (e shortcutExpander) resourceFinder(resource schema.GroupVersionResource) (schema.GroupVersionResource, bool) {
+	apiResList, err := e.discoveryClient.ServerResources()
+	if err == nil {
+		for _, apiResources := range apiResList {
+			for _, apiRes := range apiResources.APIResources {
+				if apiRes.Name == resource.Resource {
```

Comment:
@liggitt 
apiRes doesn't have Group or Version to match with -vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

It looks, only way to Group/Version match is to match with mapper which is actually happening in restmapper.go. Updated the PR.